### DISCUSSION TITLE
Custom milab-human-rna-tcr-umi-race

### DISCUSSION THREAD

**Discussion by alexell22** [milaboratory/mixcr#1567] [Q&A] [answered]
Hi!

I am trying to analyse the data from a protocol similar to milab-human-rna-tcr-umi-race, with several adjustments. Unfortunately, I have failed to modify the presets (add the sequence to the pattern before UMI). Is it possible to do this somehow?

**Comment by mizraelson** [comment]
Yes, what barcode sequence do you have?

**Comment by alexell22** [comment]
We have followed 5'RACE protocol, however, we have in R1  ^NNNNNCAGTGGTATCAACGCAGAG-UNNNNUNNNNUNNNNU-CTT, and in R2 ^NNNNNACACNNNNTTCAGGTCCTC. Really happy for help! Thanks!

**Comment by mizraelson** [comment]
As I understand your UMI is in R1. What is the NNNN sequence in the R2?

**Comment by alexell22** [comment]
Yes, it is right!  NNNNN - random nucleotides which introduced at 5’-ends of the final library for better diversity generation and cluster differentiation by Illumina sequencer. And the second NNNN in read one is STTK (listed as that sequence, but pretty random among the files).

**Comment by mizraelson** [comment, marked as answer]
I recommend trying the following command:

```
mixcr analyze generic-amplicon-with-umi \
    --species hsa \
    --rna \
    --tag-pattern "^N{5}cagtggtatcaacgcagag(UMI:N{16})ctt(R1:*)\^N{5}acacN{4}ttcaggtcctc(R2:*)" \
    --rigid-left-alignment-boundary \
    --floating-right-alignment-boundary C \
    --assemble-clonotypes-by VDJRegion \
      input_R1.fastq.gz \
      input_R2.fastq.gz \
      result
```

The `--assemble-clonotypes-by VDJRegion` parameter suggests that your reads cover the full receptor sequence. Omit this parameter if you are using short reads, for example, 150bp+150bp.
